tests/components/openerz/test_sensor.py

```diff
@@ -0,0 +1,353 @@
+"""Tests for OpenERZ component."""
+from datetime import datetime
+import unittest
+from unittest.mock import patch
+
+from requests.exceptions import RequestException
+from testfixtures import LogCapture
+
+from homeassistant.components.openerz.sensor import OpenERZSensor
+
+MOCK_DATETIME = datetime(
+    year=2019, month=12, day=10, hour=11, minute=15, second=0, microsecond=0
+)
+
+
+class MockAPIResponse:
+    """Provide fake response from the OpenERZ API."""
+
+    def __init__(self, is_ok, status_code, json_data):
+        """Initialize all the values."""
+        self.ok = is_ok
+        self.status_code = status_code
+        self.json_data = json_data
+
+    def json(self):
+        """Return response data."""
+        return self.json_data
+
+
+class TestOpenERZSensor(unittest.TestCase):
+    """Test the OpenERZ sensor."""
+
+    def setup_method(self, method):
+        """Set up things to be run when tests are started."""
+        self.mock_datetime = MOCK_DATETIME
+        self.mock_config = {
+            "platform": "openerz",
+            "name": "test_name",
+            "zip": 1234,
+            "waste_type": "glass",
+            "entity_id": "sensor.erz_glass_1234",
+        }
+
+    @patch(
+        "homeassistant.components.openerz.sensor.OpenERZSensor.update",
+        return_value=True,
+    )
+    def test_sensor_init(self, patched_update):
+        """Test whether all values initialized properly."""
+        with patch("homeassistant.components.openerz.sensor.datetime") as patched_time:
+            patched_time.now.return_value = self.mock_datetime
+
+            test_openerz = OpenERZSensor(self.mock_config)
```

Comment:
Don't create the entity directly. Instead set up the component and platform and let the platform add an entity. Then assert the state of the entity via the state machine. Make sure to patch the library api methods to avoid I/O in the tests.